docs/data-cleanup-mechanisms.md

```diff
@@ -0,0 +1,106 @@
+reason, a subscriber ends up with no subscriptions. This has happened before,
+unintentionally, see [here][sub bug] and [here][sub bug two].
+
+The [NullifySubscribersWorker] runs [every hour].
+
+[every hour]: https://github.com/alphagov/email-alert-api/blob/master/config/sidekiq.yml#L26-L28
+[NullifySubscribersWorker]: https://github.com/alphagov/email-alert-api/blob/master/app/workers/nullify_subscribers_worker.rb
+[sub bug]: https://github.com/alphagov/email-alert-api/pull/1462/commits/053859c5962eef104256661f28727b08a43e3d31
+[sub bug two]: https://github.com/alphagov/email-alert-api/pull/1462/commits/a4b3e82801d79abd3989b1dd60ffd499e7ce82ba
+
+## Archiving emails
+
+We archive emails to S3 using the [EmailArchiveWorker]. This is done to support
+long term analytics as we can query the S3 bucket through [Athena].
+
+Emails are eligible for archiving immediately after creation as long as they
+haven't already been archived (based on `archived_at` value).
+
+The [archive worker runs every hour].
+
+[Athena]: https://github.com/alphagov/email-alert-api/blob/master/docs/analytics.md#analytics-through-athena
+[EmailArchiveWorker]: https://github.com/alphagov/email-alert-api/blob/master/app/workers/email_archive_worker.rb
+[archive worker runs every hour]: https://github.com/alphagov/email-alert-api/blob/ec075c7198b965d3771726946dcd37783b9de9c5/config/sidekiq.yml#L29-L31
+
+## Deleting emails
+
+The [EmailDeletionWorker] deletes emails after they've been archived for over
+seven days. These recently archived emails get deleted regularly to keep the
+database performant as we produce a large quantity which the system only uses
+for a short period of time. In deleting these emails we also delete their
+associated subscription contents through a [db cascade].
+
+As of 05/11/2020 we generate around ~3 million emails per day.
+
+The [deletion worker runs every hour].
+
+[deletion worker runs every hour]: https://github.com/alphagov/email-alert-api/blob/ec075c7198b965d3771726946dcd37783b9de9c5/config/sidekiq.yml#L32-L34
+[EmailDeletionWorker]: https://github.com/alphagov/email-alert-api/blob/a62abc85453b723d683c2dc13f3bf0065fb86d5f/app/workers/email_deletion_worker.rb
+[db cascade]: https://github.com/alphagov/email-alert-api/blob/11fb84542e6c7f3995f419e4affaf56aa759ec6c/db/schema.rb#L206
+
+## Deleting historic data
+
+The [HistoricalDataDeletionWorker] deletes data we deem as historic everday at
+[midday]. Midday was chosen as removing historic data is deemed non urgent work
+and so running once a day, in hours, seems sufficient.
+
+In general, we remove historic, unused data after a year. One exception to this
+is that we remove subscriber lists which have been created over seven days ago
+and never had any subscriptions.
+
+We remove this data so we:
+
+* Don't pollute our analytics.
+
+* Don't run into future capacity issues.
+
+The following list explains how we deem each model's records as historic:
```

Comment:
We could just refer to the ADR for this, as it's the same content and someone can also just read the worker code, now they know it's there.